lib/logstash/inputs/tcp.rb

```diff
@@ -297,6 +304,14 @@ def ssl_context
         @ssl_context.cert_store  = load_cert_store
         @ssl_context.verify_mode = OpenSSL::SSL::VERIFY_PEER|OpenSSL::SSL::VERIFY_FAIL_IF_NO_PEER_CERT
       end
+      if ssl_supported_protocols.any?
+        min_max_version = ssl_supported_protocols.sort.map { |n| n.sub('v', '').sub('.', '_').to_sym } # 'TLSv1.2' => :TLS1_2
+        @ssl_context.min_version = min_max_version.first
+        @ssl_context.max_version = min_max_version.last
```

Comment:
as mentioned in the tcp output review, we can enable the exact list of supported protocols in the context instead of max/min, so that the behaviour is more consistent with the java context.